src/crimson/os/seastore/seastore.cc

```diff
@@ -40,10 +40,50 @@ SeaStore::SeaStore(
     transaction_manager(std::move(tm)),
     collection_manager(std::move(cm)),
     onode_manager(std::move(om))
-{}
+{
+  register_metrics();
+}
 
 SeaStore::~SeaStore() = default;
 
+void SeaStore::register_metrics()
+{
+  namespace sm = seastar::metrics;
+  using op_type_t = SeaStore::op_type_t;
+  auto lat_label = sm::label("latency");
+  std::map<op_type_t, sm::label_instance> labels_by_op_type {
+    {op_type_t::TRANSACTION,     lat_label("TRANSACTION")},
+    {op_type_t::READ,            lat_label("READ")},
+    {op_type_t::WRITE,           lat_label("WRITE")},
+    {op_type_t::GET_ATTR,        lat_label("GET_ATTR")},
+    {op_type_t::GET_ATTRS,       lat_label("GET_ATTRS")},
+    {op_type_t::STAT,            lat_label("STAT")},
+    {op_type_t::OMAP_GET_VALUES, lat_label("OMAP_GET_VALUES")},
+    {op_type_t::OMAP_LIST,       lat_label("OMAP_LIST")},
+  };
+  auto register_op_lat = [this, &labels_by_op_type](op_type_t op_type, sm::label_instance label) {
```

Comment:
the captured `labels_by_op_type` is not used.

Reply:
nit, why use a lambda for registering the metric instead of inlining it in the `for` loop below?